# LIVE CODE 3

* *Name* : Alexander Prasetyo Christianto
* *Age* : 23
* *Last Education Background* : *Electrical Engineering*
* *Occupation* : *Full Time Data Science Student Batch-001*


---

## Description

Saya adalah seorang data scientist di salah satu perusahaan asuransi. Saat ini, kamu sedang membuat model machine learning untuk memprediksi apakah klaim asuransi perjalanan seseorang diterima atau tidak berdasarkan beberapa parameter.

### Problem Statement

1. Tim Travel Insurance ingin mengetahui berapa banyak klaim asuransi diterima yang dilakukan oleh nasabah untuk setiap agensi.
2. Tim Travel Insurance ingin mengetahui tipe travel agensi apa yang paling banyak menerima klaim dan diterima.
3. Dimana negara yang paling banyak menerima klaim asuransi?
4. Apakah umur menjadi faktor yang mempengaruhi dalam melakukan klaim asuransi?
5. Berapa banyak komisi yang bisa didapatkan untuk negara tertentu seperti Singapura, Thailand, dan Malaysia?

---

## Working Station

Langkah pertama adalah melakukan *import libraries* yang akan digunakan dan melakukan koneksi ke SQL Database yang akan digunakan.

In [187]:
import mysql.connector
import pandas as pd

In [188]:
db = mysql.connector.connect(
    host="remotemysql.com",
    user="cESOsiaUiy",
    passwd="saecUeCNAL",
    database="cESOsiaUiy"
)

if db.is_connected():
    print("Koneksi sukses!")

    
cur = db.cursor()

Koneksi sukses!


#### Masalah 1

In [189]:
sql = '''
SELECT * FROM travel_insurance
'''

df = pd.read_sql_query(sql, db)

df.head(5)

id agency    agency_type distribution_channel        product_name claim  \
0   1    CBH  Travel Agency              Offline  Comprehensive Plan    No   
1   2    CBH  Travel Agency              Offline  Comprehensive Plan    No   
2   3    JZI       Airlines               Online          Value Plan    No   
3   4    JZI       Airlines               Online          Basic Plan    No   
4   5    KML  Travel Agency               Online        Premier Plan    No   

   duration    destination  net_sales  commission gender  age  
0       186       MALAYSIA        -29        9.57      F   81  
1       186       MALAYSIA        -29        9.57      F   71  
2        66  UNITED STATES       -121       42.35      F   44  
3         1       MALAYSIA        -18        6.30      M   47  
4        53         NORWAY       -130       49.40      F   48

#### Masalah 2

In [190]:
sql2 = '''
select agency, count(agency) as amount_of_claim, claim FROM travel_insurance
where claim = "Yes"
group by agency
order by amount_of_claim desc
'''

df2 = pd.read_sql_query(sql2, db)

df2

c:\Users\alexa\AppData\Local\Programs\Python\Python39\lib\site-packages\pandas\io\sql.py:761: UserWarning: pandas only support SQLAlchemy connectable(engine/connection) ordatabase string URI or sqlite3 DBAPI2 connectionother DBAPI2 objects are not tested, please consider using SQLAlchemy
  warnings.warn(


agency  amount_of_claim claim
0     C2B              547   Yes
1     LWC               36   Yes
2     JZI               31   Yes
3     KML                8   Yes
4     TTW                4   Yes
5     JWT                4   Yes
6     CCR                3   Yes
7     TST                2   Yes
8     RAB                1   Yes
9     CSR                1   Yes
10    CBH                1   Yes
11    ART                1   Yes

Agensi yang paling banyak menerima klaim adalah agensi C2B dimana jumlah klaim yang diajukan dan diterima adalah sebanyak 547 klaim.

#### Masalah 3

In [195]:
sql3 = '''
select agency_type, count(agency_type) as amount_of_accepted_claim, claim as claim_status FROM travel_insurance
where claim = "Yes"
group by agency_type
'''

df3 = pd.read_sql_query(sql3, db)

df3 

c:\Users\alexa\AppData\Local\Programs\Python\Python39\lib\site-packages\pandas\io\sql.py:761: UserWarning: pandas only support SQLAlchemy connectable(engine/connection) ordatabase string URI or sqlite3 DBAPI2 connectionother DBAPI2 objects are not tested, please consider using SQLAlchemy
  warnings.warn(


agency_type  amount_of_accepted_claim claim_status
0       Airlines                       584          Yes
1  Travel Agency                        55          Yes

Tipe agensi `Airlines` paling banyak menerima klaim yang diterima. Hal ini dapat dikarenakan orang lebih banyak menggunakan jasa penerbangan dibandingkan jasa travel darat.

#### Masalah 4

In [197]:
sql4 = '''
select destination, count(destination), claim FROM travel_insurance
where claim = "Yes"
group by destination
'''

df4 = pd.read_sql_query(sql4, db)

df4

c:\Users\alexa\AppData\Local\Programs\Python\Python39\lib\site-packages\pandas\io\sql.py:761: UserWarning: pandas only support SQLAlchemy connectable(engine/connection) ordatabase string URI or sqlite3 DBAPI2 connectionother DBAPI2 objects are not tested, please consider using SQLAlchemy
  warnings.warn(


destination  count(destination) claim
0                   SINGAPORE                 550   Yes
1                    VIET NAM                   4   Yes
2                    THAILAND                   6   Yes
3                   AUSTRALIA                   2   Yes
4                 NETHERLANDS                   2   Yes
5          KOREA, REPUBLIC OF                   5   Yes
6               UNITED STATES                  16   Yes
7                      CANADA                   2   Yes
8                   HONG KONG                   5   Yes
9                       CHINA                  12   Yes
10                      INDIA                   4   Yes
11                  INDONESIA                  10   Yes
12                NEW ZEALAND                   1   Yes
13          BRUNEI DARUSSALAM                   1   Yes
14                      JAPAN                   5   Yes
15                   MALAYSIA                   8   Yes
16  TAIWAN, PROVINCE OF CHINA                   2   Yes
17                SWITZERLAND                   1   Yes
18             UNITED KINGDOM                   1   Yes
19                     FRANCE                   2   Yes

Negara Singapura merupakan negara yang memiliki destinasi dengan jumlah klaim diterima yang terbanyak.

#### Masalah 5

In [193]:
sql5 = '''
select max(age), avg(age), min(age) from (select age, claim FROM travel_insurance
where claim = "Yes") as abc
'''

df5 = pd.read_sql_query(sql5, db)

df5

c:\Users\alexa\AppData\Local\Programs\Python\Python39\lib\site-packages\pandas\io\sql.py:761: UserWarning: pandas only support SQLAlchemy connectable(engine/connection) ordatabase string URI or sqlite3 DBAPI2 connectionother DBAPI2 objects are not tested, please consider using SQLAlchemy
  warnings.warn(


max(age)  avg(age)  min(age)
0       118   38.8294        14

Usia tertua yang melakukan klaim asuransi adalah 118 tahun, usia termuda yang melakukan klaim asuransi adalah 14 tahun dan rata-rata usia usia melakukan klaim asuransi adalah 38.82 tahun

#### Masalah 6

In [194]:
sql6 = '''
select sum(commission) as total_commision from travel_insurance
Where destination='SINGAPORE' or 
destination='THAILAND' or 
destination='MALAYSIA' and claim='Yes'
'''

df6 = pd.read_sql_query(sql6, db)

df6

c:\Users\alexa\AppData\Local\Programs\Python\Python39\lib\site-packages\pandas\io\sql.py:761: UserWarning: pandas only support SQLAlchemy connectable(engine/connection) ordatabase string URI or sqlite3 DBAPI2 connectionother DBAPI2 objects are not tested, please consider using SQLAlchemy
  warnings.warn(


total_commision
0        184437.65

---

## Poin Analisis

1. Agen C2B menerima jumlah klaim yang diterima terbanyak dibandingkan dengan agen lainnya, yaitu sejumlah 547 klaim asuransi.
2. Tipe agensi `Airlines` menerima klaim yang diterima sebanyak 584 klaim, hal ini dapat mengindikasikan banyak orang menggunakan jasa penerbangan.
3. Singapura merupakan negara yang menerima klaim asuransi diterima yang paling banyak, yaitu sejumlah 550 klaim.
4. Umur termuda yang melakukan klaim asuransi adalah 14 tahun dan umur tertua yang melakukakn klaim asuransi adalah 118 tahun dan rata-rata umur orang melakukan klaim asuransi adalah 38.82 tahun atau 38-39 tahun.
5.  Jika tim Travel Insurance hanya menerima klaim asuransi untuk destinasi negara Singapura, Thailand dan Malaysia maka jumlah klaim bisa diperoleh berdasarkan data yang sudah ada adalah sebesar 184437.65 USD.